apps/opik-backend/src/main/java/com/comet/opik/domain/DatasetItemResultMapper.java

```diff
@@ -0,0 +1,237 @@
+                .map(experimentItem -> ExperimentItem.builder()
+                        .id(UUID.fromString(experimentItem.get(0).toString()))
+                        .experimentId(UUID.fromString(experimentItem.get(1).toString()))
+                        .datasetItemId(UUID.fromString(experimentItem.get(2).toString()))
+                        .traceId(UUID.fromString(experimentItem.get(3).toString()))
+                        .input(getJsonNodeOrNull(experimentItem.get(4)))
+                        .output(getJsonNodeOrNull(experimentItem.get(5)))
+                        .feedbackScores(getFeedbackScores(experimentItem.get(6)))
+                        .createdAt(Instant.parse(experimentItem.get(7).toString()))
+                        .lastUpdatedAt(Instant.parse(experimentItem.get(8).toString()))
+                        .createdBy(experimentItem.get(9).toString())
+                        .lastUpdatedBy(experimentItem.get(10).toString())
+                        .build())
+                .toList();
+
+        return experimentItems.isEmpty() ? null : experimentItems;
+    }
+
+    static JsonNode getJsonNodeOrNull(Object field) {
+        if (null == field || StringUtils.isBlank(field.toString())) {
+            return null;
+        }
+        return JsonUtils.getJsonNodeFromString(field.toString());
+    }
+
+    static List<FeedbackScore> getFeedbackScores(Object feedbackScoresRaw) {
+        if (feedbackScoresRaw instanceof List[] feedbackScoresArray) {
+            var feedbackScores = Arrays.stream(feedbackScoresArray)
+                    .filter(feedbackScore -> CollectionUtils.isNotEmpty(feedbackScore) &&
+                            !CLICKHOUSE_FIXED_STRING_UUID_FIELD_NULL_VALUE.equals(feedbackScore.getFirst().toString()))
+                    .map(feedbackScore -> FeedbackScore.builder()
+                            .name(feedbackScore.get(1).toString())
+                            .categoryName(Optional.ofNullable(feedbackScore.get(2)).map(Object::toString)
+                                    .filter(StringUtils::isNotEmpty).orElse(null))
+                            .value(new BigDecimal(feedbackScore.get(3).toString()))
+                            .reason(Optional.ofNullable(feedbackScore.get(4)).map(Object::toString)
+                                    .filter(StringUtils::isNotEmpty).orElse(null))
+                            .source(ScoreSource.fromString(feedbackScore.get(5).toString()))
+                            .build())
+                    .toList();
+            return feedbackScores.isEmpty() ? null : feedbackScores;
+        }
+        return null;
+    }
+
+    static Map.Entry<Long, Set<Column>> groupResults(Map.Entry<Long, Set<Column>> result1,
+            Map.Entry<Long, Set<Column>> result2) {
+
+        return Map.entry(result1.getKey() + result2.getKey(), Sets.union(result1.getValue(), result2.getValue()));
+    }
+
+    static Set<Column> mapColumnsField(Map row) {
```

Comment:
Minor: I'd recommend adding Java generics to the Map input argument if possible. Maybe even consider Object.